lib/ohai/plugins/linux/platform.rb

```diff
@@ -74,12 +86,16 @@ def os_release_file_is_cisco?
       platform get_redhatish_platform(contents)
       platform_version contents.match(/(\d\.\d\.\d)/)[0]
     elsif File.exist?("/etc/redhat-release")
-      if File.exist?('/etc/os-release') && (os_release_info = os_release_file_is_cisco? ) # check if Cisco
-        platform os_release_info['ID']
-        platform_family os_release_info['ID_LIKE']
-        platform_version os_release_info['VERSION'] || ""
+      contents = File.read("/etc/redhat-release").chomp
+      if (os_release_info = os_release_file_is_cisco? ) # check if Cisco
```

Comment:
I think we could simplify this set of nested conditionals.

First, os_release_file_is_cisco? Is doing a confusing double duty as both checking for a file and also returning that data. I'd rather see something like a #get_cisco_release as the pattern of a method ending with a ? Indicates it would give use true/false. This set of conditionals here might make more sense if that was two methods. One to tell us if we are on Cisco and another to fetch the data.